architecture/2024-08-bicepdeployment-controller.md

```diff
@@ -0,0 +1,693 @@
+                "_EXPERIMENTAL_FEATURES_ENABLED": [
+                  "Extensibility"
+                ],
+                "_generator": {
+                  "name": "bicep",
+                  "version": "0.29.47.4906",
+                  "templateHash": "12114800836276126877"
+                }
+              },
+              "parameters": {
+                "location": {
+                  "type": "string"
+                },
+                "storageAccountName": {
+                  "type": "string"
+                }
+              },
+              "resources": {
+                "storageAccount": {
+                  "type": "Microsoft.Storage/storageAccounts",
+                  "apiVersion": "2021-04-01",
+                  "name": "[parameters('storageAccountName')]",
+                  "location": "[parameters('location')]",
+                  "sku": {
+                    "name": "Standard_LRS"
+                  },
+                  "kind": "StorageV2",
+                  "properties": {}
+                }
+              },
+              "outputs": {
+                "storageAccountId": {
+                  "type": "string",
+                  "value": "[resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]"
+                }
+              }
+            }
+          }
+        }
+      },
+      "outputs": {
+        "storageAccountId": {
+          "type": "string",
+          "value": "[reference('storageModule').outputs.storageAccountId.value]"
+        }
+      }
+    }
+  parameters: |
+    {
+      "storageAccountName": {
+        "value": "myaccount"
+      }
+    }
+```
+
+## Design
+
+### High Level Design
+<!--
+High level overview of the data flow and key components.
+
+Provide a high-level description, using diagrams as appropriate, and top-level
+explanations to convey the architectural/design overview. Don’t go into a lot
+of details yet but provide enough information about the relationship between
+these components and other components. Call out or highlight new components
+that are not part of this feature (dependencies). This diagram generally
+treats the components as black boxes. Provide a pointer to a more detailed
+design document, if one exists. 
+-->
+
+This design proposes the creation of two new Kubernetes custom resources and their corresponding controllers: `BicepDeployment` and `BicepDeploymentResource`. The `BicepDeployment` controller will be responsible for deploying resources defined in ARM JSON manifests to Radius, while the `BicepDeploymentResource` controller will be responsible for tracking the state of individual resources. The `BicepDeployment` controller will have a control loop that will check the status of an in-progress operation, process deletion, and process creation or update. The `BicepDeploymentResource` controller will have a similar control loop that will check the status of an in-progress operation or process deletion if necessary.
```

Comment:
**BicepDeployment** is a collection of **BicepDeploymentResource** then, right? Will the **BicepDeployment** be creating the **BicepDeploymentResource**s?